pkgs/by-name/mk/mkbootimage/package.nix

```diff
@@ -0,0 +1,41 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, libelf
+, pcre
+}:
+
+stdenv.mkDerivation {
+  pname = "mkbootimage";
+  version = "2.3-unstable-2022-05-26";
+
+  src = fetchFromGitHub {
+    owner = "antmicro";
+    repo = "zynq-mkbootimage";
+    rev = "872363ce32c249f8278cf107bc6d3bdeb38d849f";
+    hash = "sha256-5FPyAhUWZDwHbqmp9J2ZXTmjaXPz+dzrJMolaNwADHs=";
+  };
+
+  buildInputs = [ libelf pcre ];
+
+  postPatch = ''
+    substituteInPlace Makefile --replace "git rev-parse --short HEAD" "echo nix"
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm755 mkbootimage -t $out/bin
+    runHook postInstall
+  '';
+
+  hardeningDisable = [ "fortify" ];
+
+  meta = with lib; {
+    description = "An open source replacement of the Xilinx bootgen application";
+    homepage = "https://github.com/antmicro/zynq-mkbootimage";
+    license = licenses.bsd2;
+    platforms = platforms.linux;
+    maintainers = [ maintainers.fsagbuya ];
+    mainProgram = "mkbootimage";
+  };
+}
```

Comment:
```suggestion
stdenv.mkDerivation (finalAttrs: {
  pname = "mkbootimage";
  version = "2.3-unstable-2022-05-26";

  src = fetchFromGitHub {
    owner = "antmicro";
    repo = "zynq-mkbootimage";
    rev = "872363ce32c249f8278cf107bc6d3bdeb38d849f";
    hash = "sha256-5FPyAhUWZDwHbqmp9J2ZXTmjaXPz+dzrJMolaNwADHs=";
  };

  buildInputs = [ libelf pcre ];

  postPatch = ''
    substituteInPlace Makefile --replace "git rev-parse --short HEAD" "echo ${finalAttrs.src.rev}"
  '';

  installPhase = ''
    runHook preInstall
    install -Dm755 mkbootimage -t $out/bin
    runHook postInstall
  '';

  hardeningDisable = [ "fortify" ];

  meta = with lib; {
    description = "An open source replacement of the Xilinx bootgen application";
    homepage = "https://github.com/antmicro/zynq-mkbootimage";
    license = licenses.bsd2;
    platforms = platforms.linux;
    maintainers = [ maintainers.fsagbuya ];
    mainProgram = "mkbootimage";
  };
})
```

Reply:
this gives the actual git rev (just like the git command you are overwriting) Instead of substituting with some dummy value but to be honest I don't know if it changes much in the output.

Reply:
Understood. Updated the derivation.